packages/backend-common/src/reading/tree/TarArchiveResponse.ts

```diff
@@ -98,6 +98,11 @@ export class TarArchiveResponse implements ReadTreeResponse {
         }
       }
 
+      if (entry.size && entry.size >= 20000) {
```

Comment:
We can't do this for all readers, as it's used in places where you expect to get all files returned as well. We could pass the size in second `info` param in in the filter function though, allowing us to filter in the plugin.